src/app/campaign/store/actions/campaign-action.service.ts

```diff
@@ -1,106 +1,62 @@
 import { Store, select } from '@ngrx/store';
-import { Subscription } from 'rxjs';
 import { first, filter } from 'rxjs/operators';
-import * as allocationPreviewActions from './allocation-preview-action.creator';
-import * as availabilityActions from './availability-action.creator';
 import * as campaignActions from './campaign-action.creator';
-import { Flight, getFlightZoneIds, isZonesChanged, isStartAtChanged, isEndAtChanged, isInventoryChanged } from '../models';
+import * as flightPreviewActions from './flight-preview-action.creator';
+import { Flight, isZonesChanged, isStartAtChanged, isEndAtChanged, isInventoryChanged } from '../models';
 import {
   selectCampaignId,
   selectCampaignWithFlightsForSave,
   selectRoutedFlight,
-  selectRoutedLocalFlight,
   selectCampaignDoc,
-  selectCampaignAndFlights
+  selectCampaignAndFlights,
+  selectRoutedCampaignFlightDocs
 } from '../selectors';
 import { Moment } from 'moment';
 
 @Injectable()
-export class CampaignActionService implements OnDestroy {
-  private flightSub: Subscription;
-  // current flight id only to be used for checking against changed flight id to load availability on flight tab/route change
-  // for any other current/routed flight id needs, please pipe from the selectRoutedFlightId selector
-  // one reason the routed flight id selector is not used here is that is would fire on navigation before the flight is actually loaded
-  private currentFlightId: number;
+export class CampaignActionService {
+  constructor(private store: Store<any>) {}
 
-  constructor(private store: Store<any>) {
-    this.loadAvailabilityOnFlightIdChange();
-  }
-
-  ngOnDestroy() {
-    if (this.flightSub) {
-      this.flightSub.unsubscribe();
+  loadPreviewIfFlightFormChanged(formFlight: Flight, localFlight: Flight) {
+    if (this.isDateRangeValid(formFlight) && this.havePreviewParamsChanged(formFlight, localFlight)) {
+      this.loadFlightPreview({ ...formFlight, id: localFlight.id });
     }
   }
 
-  private loadAvailabilityOnFlightIdChange() {
-    this.flightSub = this.store
-      .pipe(select(selectRoutedLocalFlight))
-      .pipe(filter(flight => flight && this.currentFlightId !== flight.id))
-      .subscribe(flight => {
-        if (this.hasAvailabilityParams(flight)) {
-          this.loadAvailability(flight);
-        }
-        this.currentFlightId = flight.id;
-      });
-  }
-
-  loadAvailabilityAllocationIfChanged(formFlight: Flight, localFlight: Flight) {
-    if (this.isDateRangeValid(formFlight) && this.haveAvailabilityParamsChanged(formFlight, localFlight)) {
-      this.loadAvailability({ ...formFlight, id: localFlight.id, createdAt: localFlight.createdAt });
-      if (formFlight.totalGoal) {
-        this.loadAllocationPreview({ ...formFlight, id: localFlight.id, createdAt: localFlight.createdAt });
-      }
-    }
+  loadFlightPreview(flight: Flight) {
+    const { id: flightId, set_inventory_uri, name, startAt, endAt, zones, totalGoal, dailyMinimum } = flight;
```

Comment:
you're correct on that. I had carried the params forward unnecessarily. Refactored! I did choose to keep the Flight model itself out of the flight-preview.service because it's in core.module not in the campaign.module (even though I have not been entirely consistent about that, it's aspirational ok)